### PR TITLE
Enable sync stack for Container registry

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/ContainerRegistriesImpl.java [hunk 1]
```diff
@@ -265,291 +466,2074 @@ Mono<ContainerRegistriesGetManifestsNextResponse> getManifestsNext(
...
+     * @param payload Manifest body, can take v1 or v2 values depending on accept header.
+     * @param contentLength The Content-Length header for the request.
+     * @param contentType The manifest's Content-Type.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public void createManifestSync(
```
Comment (samvaity) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/ContainerRegistriesImpl.java:1026]: Same as above, should not be generated?
Also true for some of the API's generated below.

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/ContainerRegistriesImpl.java [hunk 2]
```diff
@@ -265,291 +466,2074 @@ Mono<ContainerRegistriesGetManifestsNextResponse> getManifestsNext(
...
+     * @param n query parameter for max number of items.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws AcrErrorsException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return list of repositories as paginated response with {@link PagedFlux}.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<String> getRepositoriesAsync(String last, Integer n) {
+        return new PagedFlux<>(
+                () -> getRepositoriesSinglePageAsync(last, n),
```
Comment (samvaity) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/ContainerRegistriesImpl.java:1204]: @srnagar This API expects a context parameter. Is that the expected contract?

File: sdk/containerregistry/azure-containers-containerregistry/pom.xml [hunk 3]
```diff
@@ -28,6 +28,8 @@
   </scm>
 
   <properties>
+    <!-- Temporarily skipping code coverage until tests are enabled/added -->
+    <jacoco.skip>true</jacoco.skip>
```
Comment (samvaity) [inline sdk/containerregistry/azure-containers-containerregistry/pom.xml:32]: https://github.com/Azure/azure-sdk-for-java/issues/32571

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java [hunk 4]
```diff
@@ -111,7 +147,35 @@ public UploadManifestResult uploadManifest(UploadManifestOptions options) {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<UploadManifestResult> uploadManifestWithResponse(UploadManifestOptions options, Context context) {
-        return this.asyncClient.uploadManifestWithResponse(options, context).block();
+        return this.uploadManifestWithResponseSync(options, context);
+    }
+
+    Response<UploadManifestResult> uploadManifestWithResponseSync(UploadManifestOptions options, Context context) {
```
Comment (lmolkova) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java:153]: please move impl into public method

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java [hunk 5]
```diff
@@ -147,7 +211,38 @@ public UploadBlobResult uploadBlob(BinaryData data) {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<UploadBlobResult> uploadBlobWithResponse(BinaryData data, Context context) {
-        return this.asyncClient.uploadBlobWithResponse(data.toByteBuffer(), context).block();
+        return this.uploadBlobWithResponseSync(data, context);
+    }
+
+    Response<UploadBlobResult> uploadBlobWithResponseSync(BinaryData data, Context context) {
```
Comment (lmolkova) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java:217]: same here, please move impl to public method

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java [hunk 6]
```diff
@@ -180,7 +275,44 @@ public DownloadManifestResult downloadManifest(DownloadManifestOptions options)
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<DownloadManifestResult> downloadManifestWithResponse(DownloadManifestOptions options, Context context) {
-        return this.asyncClient.downloadManifestWithResponse(options, context).block();
+        return this.downloadManifestWithResponseSync(options, context);
+    }
+
+    Response<DownloadManifestResult> downloadManifestWithResponseSync(DownloadManifestOptions options, Context context) {
```
Comment (lmolkova) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java:281]: and here

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java [hunk 7]
```diff
@@ -163,7 +258,7 @@ public Response<UploadBlobResult> uploadBlobWithResponse(BinaryData data, Contex
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public DownloadManifestResult downloadManifest(DownloadManifestOptions options) {
-        return this.asyncClient.downloadManifest(options).block();
+        return this.downloadManifestWithResponse(options, null).getValue();
```
Comment (lmolkova) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java:261]:
```suggestion
        return this.downloadManifestWithResponse(options, Context.NONE).getValue();
```

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java [hunk 8]
```diff
@@ -207,7 +339,36 @@ public DownloadBlobResult downloadBlob(String digest) {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<DownloadBlobResult> downloadBlobWithResponse(String digest, Context context) {
-        return this.asyncClient.downloadBlobWithResponse(digest, context).block();
+        return this.downloadBlobWithResponseSync(digest, context);
+    }
+
+    Response<DownloadBlobResult> downloadBlobWithResponseSync(String digest, Context context) {
```
Comment (lmolkova) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java:345]: and here

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java [hunk 9]
```diff
@@ -233,7 +394,28 @@ public void deleteBlob(String digest) {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> deleteBlobWithResponse(String digest, Context context) {
-        return this.asyncClient.deleteBlobWithResponse(digest, context).block();
+        return this.deleteBlobWithResponseSync(digest, context);
+    }
+
+    Response<Void> deleteBlobWithResponseSync(String digest, Context context) {
```
Comment (lmolkova) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobClient.java:400]: and here

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifact.java [hunk 10]
```diff
@@ -34,23 +53,68 @@
...
-     * @param asyncClient The async client for the given repository.
+     * @param repositoryName The name of the repository on which the service operations are performed.
+     * @param tagOrDigest The tag or digest associated with the given artifact.
+     * @param endpoint The URL string for the Azure Container Registry service.
+     * @param httpPipeline HttpPipeline that the HTTP requests and responses flow through.
+     * @param version {@link ContainerRegistryServiceVersion} of the service to be used when making requests.
      */
-    RegistryArtifact(RegistryArtifactAsync asyncClient) {
-        this.asyncClient = asyncClient;
+    RegistryArtifact(String repositoryName, String tagOrDigest, HttpPipeline httpPipeline, String endpoint, String version) {
```
Comment (lmolkova) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifact.java:73]: Please consider https://github.com/samvaity/azure-sdk-for-java/pull/4 that allows to reduce duplication

**Review by lmolkova — COMMENTED** [review]
Left some minor comments

**Review by lmolkova — APPROVED** [review]
LGTM, thanks!!!

**Pull request merged**